src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs

```diff
@@ -152,6 +152,13 @@ public string SerializePayload(string messageType, object payload, int version)
             return JsonConvert.SerializeObject(message);
         }
 
+        /// <inheritdoc/>
+        public T Clone<T>(T obj)
+        {
+            var stringObj = this.Serialize<T>(obj);
```

Comment:
Suggestion: Check for ICloneable implemention and if so use it. else fallback

Reply:
Please see https://github.com/Microsoft/vstest/pull/1649#discussion_r196123434